### PR TITLE
refactoring contract pinning

### PR DESCRIPTION
**Pull request by Aniket-Engg**

### CODE REVIEW

File: libs/remix-ui/run-tab/src/lib/actions/events.ts [hunk 1]
```diff
@@ -94,22 +96,7 @@ export const setupEvents = (plugin: RunTab, dispatch: React.Dispatch<any>) => {
...
-        for (const file of filePaths) {
-          const pinnedContract = await plugin.call('fileManager', 'readFile', file)
-          const pinnedContractObj = JSON.parse(pinnedContract)
-          if (pinnedContractObj) addPinnedInstance(dispatch, pinnedContractObj)
-        }
-      } catch(err) {
-        console.log(err)
-      }
-    }
+    await migrateSavedContracts(plugin)
```
Comment (yann300) [inline libs/remix-ui/run-tab/src/lib/actions/events.ts:99]: on the `setWorkspace` you don't load pinned contracts?

File: libs/remix-ui/run-tab/src/lib/actions/events.ts [hunk 2]
```diff
@@ -179,6 +166,50 @@ export const setupEvents = (plugin: RunTab, dispatch: React.Dispatch<any>) => {
   }, 30000)  
 }
 
+const loadPinnedContracts = async (plugin, dispatch, dirName) => {
+  console.log('loadPinnedContracts---->')
```
Comment (yann300) [inline libs/remix-ui/run-tab/src/lib/actions/events.ts:170]: should probably be removed.

File: libs/remix-ui/run-tab/src/lib/actions/events.ts (hunk 1 above)
Comment (Aniket-Engg) [reply libs/remix-ui/run-tab/src/lib/actions/events.ts:99]: yes, we load it once network is ready

File: libs/remix-ui/run-tab/src/lib/actions/events.ts (hunk 2 above)
Comment (Aniket-Engg) [reply libs/remix-ui/run-tab/src/lib/actions/events.ts]: removed

File: libs/remix-ui/run-tab/src/lib/actions/events.ts (hunk 1 above)
Comment (yann300) [reply libs/remix-ui/run-tab/src/lib/actions/events.ts:99]: yes but when the workspace change you still need to reload pinned contract no?

Comment (Aniket-Engg) [reply libs/remix-ui/run-tab/src/lib/actions/events.ts:99]: when workspace is changed, network is also reset/updated

**Review by yann300 — APPROVED** [review]

**Pull request merged**